src/MSBuild/MSBuild.csproj

```diff
@@ -330,4 +330,14 @@
 
   </Target>
 
+  <UsingTask TaskName="ValidateMSBuildPackageDependencyVersions" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <Task>
+      <Code Source="ValidateMSBuildPackageDependencyVersions.cs" Language="cs" />
+    </Task>
+  </UsingTask>
+
+  <Target Name="ValidateMSBuildPackageDependencyVersions" BeforeTargets="AfterBuild">
```

Comment:
Only run this target on .NET Framework builds (since that's the only case where the binding redirects apply).